src/main/java/org/folio/fqm/service/PermissionsRegularService.java

```diff
@@ -27,36 +27,43 @@
 public class PermissionsRegularService implements PermissionsService {
 
   // package-private for visibility in unit tests
-  @Value("${folio.is-eureka:false}")
+
+  @Value("${folio.is-eureka}")
   boolean isEureka;
 
   @Value("${mod-fqm-manager.permissions-cache-timeout-seconds:60}")
   private long cacheDurationSeconds;
 
   private final FolioExecutionContext context;
   private final ModPermissionsClient modPermissionsClient;
-  private final Cache<TenantUserPair, Set<String>> cache = Caffeine.newBuilder().expireAfterWrite(cacheDurationSeconds, TimeUnit.SECONDS).build();
+  private final ModRolesKeyClockClient modRolesKeycloakClient;
+  private final Cache<TenantUserPair, Set<String>> cache = Caffeine.newBuilder()
+    .expireAfterWrite(cacheDurationSeconds, TimeUnit.SECONDS)
+    .build();
   private final EntityTypeFlatteningService entityTypeFlatteningService;
+
   private static final List<String> CROSS_TENANT_FQM_PERMISSIONS = List.of(
     "fqm.entityTypes.item.get",
     "fqm.query.async.results.get",
     "fqm.query.async.post"
   );
 
+  @Override
   public Set<String> getUserPermissions() {
     return getUserPermissions(context.getTenantId());
   }
 
   public Set<String> getUserPermissions(String tenantId) {
     TenantUserPair key = new TenantUserPair(tenantId, context.getUserId());
-    return cache.get(key, k -> isEureka ? getUserPermissionsFromRolesKeycloak(k.userId()) : getUserPermissionsFromModPermissions(tenantId, k.userId()));
+    return cache.get(key, k -> isEureka ? getUserPermissionsFromRolesKeycloak(tenantId, k.userId()) : getUserPermissionsFromModPermissions(tenantId, k.userId()));
```

Comment:
Nit: Get the tenant ID from `k` instead of using `tenantId`. That way, the lambda function doesn't have to be instantiated every time the method is called (if you reference something outside the lambda, then the lambda's has to be a closure that covers whatever scope the thing lives in. So by referencing `isEureka`, the lambda is effectively an instance variable that and only needs to be instantiated once per instance. By referencing a method parameter, the closure has a method scope, so it needs instantiated for every method call)

Reply:
Done